api/base/exceptions.py

```diff
@@ -80,20 +88,88 @@ class Conflict(APIException):
     default_detail = ('Resource identifier does not match server endpoint.')
 
 
-class InvalidQueryStringError(JSONAPIException):
+class JSONAPIParameterException(JSONAPIException):
+    def __init__(self, detail=None, parameter=None):
+        source = {
+            'parameter': parameter
+        }
+        super(JSONAPIParameterException, self).__init__(detail=detail, source=source)
+
+
+class JSONAPIAttributeException(JSONAPIException):
+    def __init__(self, detail=None, attribute=None):
+        source = {
+            'pointer': '/data/attributes/{}'.format(attribute)
+        }
+        super(JSONAPIAttributeException, self).__init__(detail=detail, source=source)
+
+
+class InvalidQueryStringError(JSONAPIParameterException):
     """Raised when client passes an invalid value to a query string parameter."""
     default_detail = 'Query string contains an invalid value.'
     status_code = http.BAD_REQUEST
 
-    def __init__(self, detail=None, parameter=None):
-        super(InvalidQueryStringError, self).__init__(detail=detail, source={
-            'parameter': parameter
-        })
 
+class InvalidFilterOperator(JSONAPIParameterException):
+    """Raised when client passes an invalid operator to a query param filter."""
+    status_code = http.BAD_REQUEST
+
+    def __init__(self, detail=None, value=None, valid_operators=None):
+        if value and not detail:
+            valid_operators = valid_operators or ['eq', 'lt', 'lte', 'gt', 'gte', 'contains', 'icontains']
+            detail = "Value '{0}' is not a supported filter operator; use one of {1}.".format(
+                value,
+                valid_operators
+            )
+        super(InvalidFilterOperator, self).__init__(detail=detail, parameter='filter')
+
+
+class InvalidFilterValue(JSONAPIParameterException):
+    """Raised when client passes an invalid value to a query param filter."""
+    default_detail = 'Invalid filter value'
+    status_code = http.BAD_REQUEST
+
+    def __init__(self, detail=None, value=None, field_type=None):
+        if not detail:
+            detail = "Value '{0}' is not valid".format(value)
+            if field_type:
+                detail += " for a filter on type {0}".format(
+                    field_type
+                )
+            detail += "."
+        super(InvalidFilterValue, self).__init__(detail=detail, parameter='filter')
+
+
+class InvalidFilterError(JSONAPIParameterException):
+    """Raised when client passes an malformed filter in the query string."""
+    default_detail = 'Query string contains a malformed filter.'
+    status_code = http.BAD_REQUEST
+
+    def __init__(self, detail=None):
+        super(InvalidFilterError, self).__init__(detail=detail, parameter='filter')
+
+
+class InvalidFilterComparisonType(JSONAPIAttributeException):
```

Comment:
Ah ok, I see the rationale. However, spec seems to reserve `pointer` for the JSON request body:

> pointer: a JSON Pointer [RFC6901] to the associated entity in the request document [e.g. "/data" for a primary data object, or "/data/attributes/title" for a specific attribute].

..though it's a little interpretation. "parameter" is a little less ambiguous:

> parameter: a string indicating which URI query parameter caused the error.

http://jsonapi.org/format/#error-objects